### PR TITLE
feat(optimizer): Add Container Backend for local hyperparameter optimization

### LINKED ISSUE #153: Support Local Execution for Optimizer
### What you would like to be added?

With the successful release of local execution mode for the `TrainingClient`, it would be great to extend this capability to the `OptimizerClient` for optimizing TrainJobs locally without requiring Kubernetes.

```python
from kubeflow.optimizer import OptimizerClient, ContainerBackendConfig, Search, TrialConfig

client = OptimizerClient(backend_config=ContainerBackendConfig())

# Create OptimizationJob with the same template
optimization_id = client.optimize(
    trial_template=template,
    trial_config=TrialConfig(num_trials=10, parallel_trials=2),
    search_space={
        "learning_rate": Search.loguniform(0.001, 0.1),
        "num_epochs": Search.choice([5, 10, 15]),
    },
)

print(f"OptimizationJob created: {optimization_id}")

```

Running optimization jobs locally would help Data Scientists quickly prototype and validate their tuning strategies before deploying to production without needing to deal with cluster complexity

### Why is this needed?

Delivering local execution mode across different clients in Kubeflow SDK, including for optimization models will significantly improve usability.

### Love this feature?

Give it a 👍 We prioritize the features with most 👍

### PR DESCRIPTION
**Pull request by NarayanaSabari**
**Fixes #153**

## Summary

Implements Container Backend for OptimizerClient to enable local hyperparameter optimization without Kubernetes. Uses Optuna TPE for adaptive sampling and Docker/Podman for trial execution.

### Core Files

**`kubeflow/optimizer/backends/container/backend.py`** (~500 lines)
- Implements `OptimizerBackend` interface
- Integrates Optuna TPE sampler with SQLite persistence
- Uses `TrainerClient` Container Backend for trial execution
- Parallel execution via `ThreadPoolExecutor`
- Metric extraction from container logs via regex

**`kubeflow/optimizer/backends/container/storage.py`** (~200 lines)
- Local JSON file persistence for job/trial metadata
- File structure: `{storage_path}/{job_name}/experiment.json` + `trials/*.json`

**`kubeflow/optimizer/backends/container/types.py`** (~100 lines)
- `ContainerBackendConfig` with validation
- Parameters: `storage_path`, `max_parallel_trials`, `pull_policy`, `container_runtime`

### API Changes

**`kubeflow/optimizer/__init__.py`**
- Added export: `ContainerBackendConfig`
- No breaking changes to existing APIs

### Examples & Documentation

- `examples/optimizer/simple-local-example.py` - Working example (3 trials, tested)
- `examples/optimizer/LOCAL_OPTIMIZATION_GUIDE.md` - Technical guide
- `kubeflow/optimizer/backends/container/README.md` - Implementation details

## Usage

```python
from kubeflow.optimizer import OptimizerClient, ContainerBackendConfig, Search, TrialConfig, Objective

client = OptimizerClient(backend_config=ContainerBackendConfig(
    max_parallel_trials=3
))

job_name = client.optimize(
    trial_template=template,
    search_space={"learning_rate": Search.loguniform(0.001, 0.1)},
    objectives=[Objective(metric="accuracy", direction="maximize")],
    trial_config=TrialConfig(num_trials=10, parallel_trials=3),
)

results = client.get_best_results(job_name)
```

## Key Design Decisions

1. **Optuna Integration** - Mature TPE algorithm, SQLite persistence, resume support
2. **TrainerBackend Reuse** - Leverages existing container orchestration for trials
3. **JSON Storage** - Simple, debuggable state persistence (vs K8s CRDs)
4. **ThreadPoolExecutor** - Simple parallelism without distributed coordination

## Testing

- ✅ End-to-end tested with 3 and 10 trial runs
- ✅ Sequential and parallel execution validated
- ✅ Metric extraction working (`metric_name: value` pattern)
- ✅ Job lifecycle complete: create → optimize → results → cleanup

## Dependencies

- `optuna>=3.0.0` (new, for hyperparameter optimization)
- `docker>=6.0.0` (existing, via `[docker]` extras)

## Breaking Changes

None. New backend addition only.

## Removal Files
- Added some README files for reviewing, after review we can delete all of those.

### CODE REVIEW

File: examples/optimizer/simple-local-example.py
Comment (aniketpati1121) [inline examples/optimizer/simple-local-example.py]: Please remove emojis and decorative console prints across the file.
Keep output and log messages simple and consistent with other SDK modules.
